pkg/components/prometheus-operator/component_test.go

```diff
@@ -98,3 +127,75 @@ component "prometheus-operator" {
 		})
 	}
 }
+
+//nolint:funlen
+func TestConversion(t *testing.T) {
+	testCases := []struct {
+		Name                 string
+		inputConfig          string
+		expectedManifestName string
+		expected             string
+		jsonPath             string
+	}{
+		{
+			Name: "use external_url param",
+			inputConfig: `
+component "prometheus-operator" {
+  prometheus {
+    external_url = "https://prometheus.externalurl.net"
+  }
+}
+`,
+			expectedManifestName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			expected:             "https://prometheus.externalurl.net",
+			jsonPath:             "{.spec.externalUrl}",
+		},
+		{
+			Name: "no external_url param",
+			inputConfig: `
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			expectedManifestName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			expected:             "https://prometheus.mydomain.net",
+			jsonPath:             "{.spec.externalUrl}",
+		},
+		{
+			Name: "ingress creation for prometheus",
+			inputConfig: `
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			expectedManifestName: "prometheus-operator/templates/prometheus/ingress.yaml",
+			expected:             "prometheus.mydomain.net",
+			jsonPath:             "{.spec.rules[0].host}",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			component := newComponent()
+			m := testutil.RenderManifests(t, component, name, tc.inputConfig)
+			gotConfig := testutil.ConfigFromMap(t, m, tc.expectedManifestName)
+
+			testutil.MatchJSONPathStringValue(t, gotConfig, tc.jsonPath, tc.expected)
```

Comment:
It will be fresh one because we are calling `newComponent` unlike some tests where we extract the component object from the global map of components. 

About putting it into a separate helper to avoid copying these three lines, I think lets wait until there are multiple of such places where we see it becoming very painful to maintain.

Reply:
The idea is to re-use this code for multiple components from the very beginning. I don't understand why we can't make it easy to use for other components from the first version. It seems to me just like deferring this work to the person adding tests for other components...

Reply:
Well, the same argument is true for not using `reflect.Value`. Why can't we use it right now it is like deferring the work to next person adding tests.

Also the problem with adding a helper here is that now we just don't have to pass component, but we also need to pass the name, the expected manifest name, json path and expected value. We can solve it by using struct like you mentioned before, dump everything in the struct and send that. But like I said we decided to revisit the struct vs reflect.Value later so I will defer this as well.